src/Assert.php

```diff
@@ -1940,6 +1941,42 @@ public static function isMap($array, $message = '')
         }
     }
 
+    /**
+     * @param Closure $closure
+     * @param string $message
+     * @return void
+     * @throws InvalidArgumentException
+     */
+    public static function isStatic($closure, $message = '')
+    {
+        Assert::isCallable($closure);
+        $reflection = new ReflectionFunction($closure);
```

Comment:
This is not type safe, because:

- The argument `$closure` is not typed, except with a docblock
- The docblock says `Closure`, but is enforced with `isCallable`
- The assertion `isCallable` uses `is_callable()`, which has type `callable|mixed`
- But ReflectionFunction accepts `Closure|string`

End result, the input type is enforced as `callable` and then passed to a constructor that accepts `Closure|string`. I believe the correct solution here is to type cast:

```php

```php
if (($closure instanceof Closure) === false) {
    $closure = Closure::fromCallable($closure);
}
```

Reply:
Fixed by just type hinting as that made the most sense to me. It also then matches the throws assertion.